cpp/src/gandiva/gdv_function_stubs_test.cc

```diff
@@ -514,7 +514,31 @@ TEST(TestGdvFnStubs, TestInitCap) {
   EXPECT_FALSE(ctx.has_error());
 
   out_str = gdv_fn_initcap_utf8(ctx_ptr, "{ÕHP,pqśv}Ń+", 15, &out_len);
-  EXPECT_EQ(std::string(out_str, out_len), "{ÕHP,pqśv}Ń+");
+  EXPECT_EQ(std::string(out_str, out_len), "{Õhp,Pqśv}Ń+");
+  EXPECT_FALSE(ctx.has_error());
+
+  out_str = gdv_fn_initcap_utf8(ctx_ptr, "sɦasasdsɦsd\"sdsdɦ", 19, &out_len);
+  EXPECT_EQ(std::string(out_str, out_len), "Sɦasasdsɦsd\"Sdsdɦ");
+  EXPECT_FALSE(ctx.has_error());
+
+  out_str = gdv_fn_initcap_utf8(ctx_ptr, "mysuperscipt@number²isfine", 27, &out_len);
+  EXPECT_EQ(std::string(out_str, out_len), "Mysuperscipt@Number²Isfine");
+  EXPECT_FALSE(ctx.has_error());
+
+  out_str = gdv_fn_initcap_utf8(ctx_ptr, "Ő<tŵas̓老ƕɱ¢vIYwށ", 25, &out_len);
+  EXPECT_EQ(std::string(out_str, out_len), "Ő<Tŵas̓老Ƕɱ¢Viywށ");
```

Comment:
why did the 25 disappear in the output?

Reply:
Also looks v is captialised. Is ¢ treated as space character?
Btw since we are keeping the behaviour same as oracle, did you make sure to match these expected values in unittests by running them on oracle?

Reply:
About the first question, I think it is an error in Github to display some characters. I will send a print of how the code is being displayed in the IDE:
![print_string](https://user-images.githubusercontent.com/47724385/124675846-daf54d00-de93-11eb-9694-7ae4d7c9c115.png)

Reply:
About the second question. Yes, the cent symbol is treated as a space character, since its character group is for symbol and currency and the function considers every character, except letters and numbers, as space. [Here is](https://github.com/apache/arrow/blob/9a2c010706f1822cac3156a3b594206f07f656e6/cpp/src/gandiva/gdv_function_stubs.cc#L606) the method that applies the logic I mentioned.

Reply:
About testing, I ran all unit tests cases in the [Oracle Live SQL](https://livesql.oracle.com/apex/f?p=590:1000) tool on the web to compare the response and [that tool](https://onlineutf8tools.com/generate-random-utf8) to generate random strings in some tests.